daemon/daemon.go

```diff
@@ -782,9 +782,11 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 		return nil, fmt.Errorf("Couldn't restore custom images: %s", err)
 	}
 
+	migrationStart := time.Now()
 	if err := v1.Migrate(config.Root, d.driver.String(), d.layerStore, d.imageStore, tagStore, distributionMetadataStore); err != nil {
 		return nil, err
 	}
+	logrus.Debugf("migration took %.2f seconds", time.Since(migrationStart).Seconds())
```

Comment:
It might be good to be more specific in this message to specify what's being migrated and why.

And I'd suggest putting migration log messages at a different log level than "debug", so people can more readily see what's going on.